src/Engine/ProtoCore/Lang/BuiltInMethods.cs

```diff
@@ -797,6 +804,7 @@ public BuiltInMethods(Core core)
                         new KeyValuePair<string, ProtoCore.Type>("object", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.Var, 0)),
                     },
                     ID = BuiltInMethods.MethodID.ToStringFromObject,
+                    MethodAttributes = new MethodAttributes(true, false, @"This method is obsolete, please use 'ToStringFromObjectAndFormat' "),
```

Comment:
Can we make the existing `FromObject` nodemodel node point to your new builtin function instead (by passing in the old default for format specifier) and remove this builtin function altogether? Same question for the `FromArray` nodemodel node.

Reply:
oh.. hmm, I think I tried something slightly different, my attempt was to add ports to the existing nodes, though I guess we could try injecting the value and not messing with ports.

But - this will mean that existing ds code that calls `__StringFromObject` will no longer work. I don't know if that is a problem or not, but I thought it could be.

Reply:
Do you mean in code block nodes or existing user-defined DS files? I don't remember if these functions with `__` can be called from CBNs.
I'm sure we'll need to update existing language tests but don't expect many cases of the builtin functions used directly by users.

Reply:
1. you can definitely call built-ins from codeblocks.
2. If this counts as a public function or not, I have no idea, `__` to me implies internal methods, but we've never communicated that to users - and because I want this PR to be as low risk as possible, I don't wish to remove them or mess with them, I want to introduce new nodes and mark them as experimental somehow to avoid accumulating more technical/API debt.
3. @Amoursol @jasonstratton - the sooner we can define an API boundary IMO the better to avoid these conversations and inconsistency for users.
![image](https://github.com/user-attachments/assets/fef01492-9a60-4c2f-a37d-3161558348e3)

Reply:
Yes, they are and were meant to be internal functions. They were not meant to be publicly exposed to users, which is why they are wrapped as other nodes. Yes, they could have been exposed in code block nodes inadvertently, which may have been an oversight or carelessness on the Dynamo team but having said that IMO they can be safely removed to remove tech debt. I just wanted to let you know that I'm not asking you to remove them, that's up to you. I'm sorry if you don't like having these conversations but I will let folks know my opinion as long as I'm reviewing PRs.

Reply:
@aparajit-pratap - appreciate your opinion here on this particular issue of course - 

what I meant about "having these conversations" - was specifically that I want to get us to a place where we don't **need** to have a conversation each time we want to remove a method. It should be mechanical because something is either public and relied upon, or it's not. We intended to do this as a group after the 3.0 release - but have not yet done so formally.

I think it will be a boon to consistency for users and productivity for developers to get to this point.